fulp_modules/main_features/bloodsuckers/code/bloodsucker/bloodsucker_crypt.dm

```diff
@@ -602,6 +613,100 @@
 			return
 
 
+/obj/structure/bloodsucker/vassalrack/proc/shapeshift_victim(mob/living/carbon/human/user, mob/living/carbon/human/target)
+	var/datum/antagonist/bloodsucker/bloodsuckerdatum = user.mind.has_antag_datum(/datum/antagonist/bloodsucker)
+
+	if(target.stat == DEAD)
+		to_chat(user, "<span class='notice'>Do you wish to rebuild this body as something greater? This will remove any restraints they might have, make them into a mute husk, and will cost 150 Blood!</span>")
+		var/list/revive_options = list(
+			"Yes" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_yes"),
+			"No" = image(icon = 'icons/hud/radial.dmi', icon_state = "radial_no")
+			)
+		var/revive_response = show_radial_menu(user, src, revive_options, radius = 36, require_near = TRUE)
+		switch(revive_response)
+			if("Yes")
+				to_chat(user, "<span class='danger'>You have brought [target] back from the dead!</span>")
+				to_chat(target, "<span class='announce'>As blood drips over your body, your heart begins to beat... You live again!</span>")
+				user.blood_volume -= 150
+				target.revive(full_heal = TRUE, admin_revive = TRUE)
```

Comment:
will this aheal restore their blood level, allowing you to drain them again?

... can bloodsuckers drain husks?

Reply:
Husks shouldn't have blood, because they're husks.

Reply:
but they do

Reply:
only changeling husks dont have blood iirc